File: lua/bob/bob.lua

return {
    hey = function(str)
        -- Empty string.
        if not str or #str == 0 then
            return "Fine, be that way."

        -- Question mark present.
        elseif str:find("%?") then
            return "Sure"

        -- Two or more repetitions of uppercase.
        elseif str:find("%u%u") then
            return "Whoa, chill out!"

        -- Not caught by another rule.
        else
            return "Whatever"
        end
    end,
}

File: julia/bob/bob.jl

pattern_uppercase = r"[A-Z]+"
pattern_question = r"\?\s*$"
pattern_empty = r"^\s*$"

function bob(stimulus::AbstractString)
    if occursin(pattern_empty, stimulus)
        "Fine. Be that way!"
    elseif occursin(pattern_uppercase, stimulus) && uppercase(stimulus) == stimulus
        if occursin(pattern_question, stimulus)
            "Calm down, I know what I'm doing!"
        else
            "Whoa, chill out!"
        end
    elseif occursin(pattern_question, stimulus)
        "Sure."
    else
        "Whatever."
    end
end
